From e16ad6f76fbb4cc25c2cb2e4b18a65152fc2bd33 Mon Sep 17 00:00:00 2001
From: Barna Zsombor Klara <zsombor.klara@cloudera.com>
Date: Fri, 25 Nov 2016 10:51:36 +0800
Subject: [PATCH 0862/1164] HIVE-15168: Flaky test:
 TestSparkClient.testJobSubmission (still flaky)
 (Barna Zsombor Klara via Rui Li, reviewed by
 Xuefu Zhang and Rui Li)

Change-Id: Ide04845ba478f350fe04e527ab38d603c9753a58
---
 .../apache/hive/spark/client/SparkClientImpl.java  |    1 +
 .../apache/hive/spark/client/TestSparkClient.java  |    1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java b/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
index 56a3ae4..165e361 100644
--- a/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
+++ b/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
@@ -532,6 +532,7 @@ private void redirect(String name, Redirector redirector) {
         @Override
         public void operationComplete(io.netty.util.concurrent.Future<Void> f) {
           if (f.isSuccess()) {
+            // If the spark job finishes before this listener is called, the QUEUED status will not be set
             handle.changeState(JobHandle.State.QUEUED);
           } else if (!promise.isDone()) {
             promise.setFailure(f.cause());
diff --git a/spark-client/src/test/java/org/apache/hive/spark/client/TestSparkClient.java b/spark-client/src/test/java/org/apache/hive/spark/client/TestSparkClient.java
index 344579c..406a015 100644
--- a/spark-client/src/test/java/org/apache/hive/spark/client/TestSparkClient.java
+++ b/spark-client/src/test/java/org/apache/hive/spark/client/TestSparkClient.java
@@ -91,7 +91,6 @@ public void call(SparkClient client) throws Exception {
         // state changes.
         assertFalse(((JobHandleImpl<String>)handle).changeState(JobHandle.State.SENT));
 
-        verify(listener).onJobQueued(handle);
         verify(listener).onJobStarted(handle);
         verify(listener).onJobSucceeded(same(handle), eq(handle.get()));
       }
-- 
1.7.9.5

